File: Arrays & Strings/28. Find the Index of the First Occurrence in a String.cpp
#include <bits/stdc++.h>


class Solution {
public: 
    int strStr(std::string haystack, std::string needle) {
        int m = haystack.length(), n = needle.length();
        for (int i = 0; i < m - n + 1; i++) {
            if (haystack.substr(i, n) == needle) return i;
        }   
        return -1;
    }
};


int main() {
    std::string haystack, needle;
    std::cin >> haystack >> needle;

    Solution sol;
    std::cout << sol.strStr(haystack, needle) << std::endl;

    return 0;
}

File: Arrays & Strings/28. Find the Index of the First Occurrence in a String.py
"""28. Find the Index of the First Occurrence in a String
"""


class Solution:
    """Solution Class
    """

    def strStr(self, haystack: str, needle: str) -> int:
        """strStr
        Function to find index of the first occurance in a string

        Args:
            haystack (str): string
            needle (str): target substring

        Returns:
            int: index of the first occurance if found or -1 
        """

        len_h: int = len(haystack)
        len_n: int = len(needle) 
        
        if len_h < len_n:
            return -1
        
        for i in range(len_h): 
            if haystack[i: i + len_n] == needle:
                return i

        return -1


def main() -> None:
    sol = Solution()

    haystack = "sadbutsad"
    needle = "sad"

    print(sol.strStr(haystack, needle))


if __name__ == "__main__":
    main()
